src/graphql/device/Resolvers.js

```diff
@@ -0,0 +1,198 @@
+const axios = require('axios');
+const UTIL = require('../utils/AxiosUtils');
+const LOG = require('../../utils/Log');
+
+const params = {
+  token: null,
+};
+const setToken = ((token) => {
+  params.token = token;
+});
+const optionsAxios = ((method, url) => UTIL.optionsAxios(method, url, params.token));
+
+const Resolvers = {
+  Query: {
+
+    async getDeviceById(root, { deviceId }, context) {
+      setToken(context.token);
+
+      const device = {};
+
+      try {
+        const { data: deviceData } = await axios(optionsAxios(UTIL.GET, `/device/${deviceId}`));
+        device.id = deviceData.id;
+        device.label = deviceData.label;
+        device.attrs = [];
+        Object.keys(deviceData.attrs).forEach((key) => {
+          for (let i = 0; i < deviceData.attrs[key].length; i += 1) {
+            let valueType = formatValueType(deviceData.attrs[key][i].value_type);
+
+            device.attrs.push({
+              label: deviceData.attrs[key][i].label,
+              type: valueType,
+            });
+          }
+        });
+
+        return (device);
+      } catch (err) {
+        LOG.error(err);
+      }
+    },
+
+    async getDevices(root, params) {
+      // building the request string
+      let requestParameters = {
+        page_size: params.page['size'] || 20,
+        page_num: params.page['number'] || 1,
+      }
+      if (params.hasOwnProperty('filter') && params.filter.hasOwnProperty('label') ){
+        requestParameters.label = params.filter.label;
+      }
+      let requestString = '/device?';
+      const keys = Object.keys(requestParameters);
+      const last = keys[keys.length - 1];
+      keys.forEach((element) => {
+        if (element === last) {
+          requestString += `${element}=${requestParameters[element]}`;
+        } else {
+          requestString += `${element}=${requestParameters[element]}&`;
+        }
+      });
+
+      try {
+        const { data: fetchedData } = await axios(optionsAxios(UTIL.GET, requestString));
+        const deviceList = ([{
+          totalCount: fetchedData.devices.length,
+          hasNextPage: fetchedData.pagination.has_next,
+          hasPreviousPage: fetchedData.pagination.page != 1,
+          devices: fetchedData.devices,
+        }]);
+
+        return deviceList;
+
+      } catch (error) {
+        LOG.error(error);
+      }
+    },
+
+    getDeviceHistory(root, params){
+      const history = [];
+      const keys = Object.keys(params.input);
+      keys.shift();
+      const requestStringPt1 = '/history/device/';
+      let requestStringPt2 = '/history';
+      
+      if (keys.length != 0) {
+        requestStringPt2 += '?';
+        const lastKey = keys[keys.length - 1];
+        keys.forEach((element) => {
+          if (element === lastKey) {
+            requestStringPt2 += `${element}=${params[element]}`;
```

Comment:
Note: When the history api receives more than one attr, the history has a different return.

Reply:
Note 2: When there is still no history for that attr and only the history of an attr is requested, the api returns 404.